apache2/msc_xml.c

```diff
@@ -114,23 +277,44 @@ int xml_complete(modsec_rec *msr, char **error_msg) {
     *error_msg = NULL;
 
     /* Only if we have a context, meaning we've done some work. */
-    if (msr->xml->parsing_ctx != NULL) {
-        /* This is how we signalise the end of parsing to libxml. */
-        xmlParseChunk(msr->xml->parsing_ctx, NULL, 0, 1);
+    if (msr->xml->parsing_ctx != NULL || msr->xml->parsing_ctx_arg != NULL) {
+        if (msr->xml->parsing_ctx != NULL &&
+            msr->txcfg->parse_xml_into_args != MSC_XML_ARGS_ONLYARGS) {
```

Comment:
This will also apply to `MSC_XML_ARGS_OFF`. Is that correct?

Reply:
Yes, same as [this](https://github.com/owasp-modsecurity/ModSecurity/pull/3358#discussion_r2061536318) comment.